### PR TITLE
Fix watch to emit an error twice on error. Add tests.

### LINKED ISSUE #392: watch with resourceVersion too old triggers done() callback with null err
It appears that when the resourceVersion is no longer available in the cluster we probably are receiving a `410 GONE` response. However I'm observing that the `done()` callback has a `null` value for the error.

Any way we can more sanely handle this scenario?

Thanks!

### PR DESCRIPTION
**Pull request by brendandburns**
Fixes #392

### CODE REVIEW

File: src/watch_test.ts [hunk 1]
```diff
@@ -168,7 +169,11 @@ describe('Watch', () => {
         expect(receivedObjects).to.deep.equal([obj1.object]);
 
         expect(doneCalled).to.equal(true);
-        expect(doneErr).to.deep.equal(errIn);
+        expect(doneErr.length).to.equal(2);
+        if (doneErr.length == 2) {
```
Comment (drubin) [inline src/watch_test.ts]: This `if check` is redundant as the first check already guards against it.

**Comment by drubin** [comment]
@brendandburns the linting is off but it looks good.

File: src/watch_test.ts (hunk 1 above)
Comment (brendandburns) [reply src/watch_test.ts]: done.

**Comment by brendandburns** [comment]
lint fixed, comment addressed. Please re-check.

Thanks!

**Pull request merged**